cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -1006,6 +1080,15 @@ const FunctionDoc iso_week_doc{
      "cannot be found in the timezone database."),
     {"values"}};
 
+const FunctionDoc week_doc{
+    "Extract week of year number",
+    ("First week has the majority (4 or more) of its days in January.\n"
+     "Year can have 52 or 53 weeks. Week numbering can start with 0 or 1 using "
+     "DayOfWeekOptions.one_based_numbering.\n"
+     "Returns an error if timestamp has a defined timezone. Null values return null."),
```

Comment:
Why does not work for timezone data? (or is this a copy-paste left-over?)

Reply:
Leftover from before this would generally be possible. Removed.